### PR TITLE
Remove `PyGitHub` from default test setup env

### PR DESCRIPTION
**Pull request by scbedd**
While resolving #33722 , I am realizing that building `pynacl` for a windows wheel is a nightmare and a half. Given that we don't actually use `pygithub` except for creating issues or generating a packaging update for mgmt packages, we don't need it always present.

I am moving `pygithub` to a extras dep in azure-sdk-tools. In addition, I am updating `update_pr.yml` to use the local version of azure-sdk-tools, versus recloning a repo we are **already in**.

This should allow me to bypass this install issue that is present on `pypy39` in #33722 

@msyyc sir can you please tell me if this update of update_pr (to install local azure-sdk-tools) breaks anything? It's currently set so that it reclones the repo we are already on, just from `main`. Given that your autogenerated branches are based off of latest `main`, I'm not certain this is actually necessary.

Let me know please!

### CODE REVIEW

**Comment by msyyc** [comment]
@scbedd I test your PR locally and it doesn't influence our release pipeline. Approved

**Comment by scbedd** [comment]
Ok. I'll update the PR to not fail, then merge the changes. Thank you!

File: tools/azure-sdk-tools/setup.py [hunk 1]
```diff
@@ -60,6 +58,7 @@
         ":python_version<'3.11'": ["tomli==2.0.1"],
         "build": ["six", "setuptools", "pyparsing", "certifi", "cibuildwheel"],
         "conda": ["beautifulsoup4"],
-        "systemperf": ["aiohttp>=3.0", "requests>=2.0", "tornado==6.0.3", "httpx>=0.21", "azure-core"]
+        "systemperf": ["aiohttp>=3.0", "requests>=2.0", "tornado==6.0.3", "httpx>=0.21", "azure-core"],
+         "ghtools": ["PyGithub>=1.59.0"],
```
Comment (mccoyp) [inline tools/azure-sdk-tools/setup.py:62]:
```suggestion
        "ghtools": ["PyGithub>=1.59.0"],
```

**Review by mccoyp — APPROVED** [review]
LGTM! Just one whitespace blip

File: tools/azure-sdk-tools/setup.py (hunk 1 above)
Comment (scbedd) [reply tools/azure-sdk-tools/setup.py]: (╯°□°)╯︵ ┻━┻

**Comment by scbedd** [comment]
Closing. This PR is supplanted by #33722

**Pull request closed without merging**